src/compiler/moduleSpecifiers.ts

```diff
@@ -996,7 +996,7 @@ function tryGetModuleNameFromRootDirs(rootDirs: readonly string[], moduleFileNam
 }
 
 function tryGetModuleNameAsNodeModule({ path, isRedirect }: ModulePath, { getCanonicalFileName, canonicalSourceDirectory }: Info, importingSourceFile: SourceFile, host: ModuleSpecifierResolutionHost, options: CompilerOptions, userPreferences: UserPreferences, packageNameOnly?: boolean, overrideMode?: ResolutionMode): string | undefined {
-    if (!host.fileExists || !host.readFile) {
+    if (!host.readFile) {

```

Comment:
On the `ModuleSpecifierResolutionHost`, `host.fileExists` is always defined, but `host.readFile` is optionally defined. 

Searching the codebase for `host.fileExists` showed it being used multiple times without a check, so I felt safe removing this. If you think this is wrong or risky, I'm happy to replace this with 

```
if (host.fileExists === undefined || !host.readFile)
```
 or however you think the codebase is best maintained.

Reply:
This one I believe is correct; this same change was made in #53463.